Closes #16 16_adtr: ADTR vignette

bundfussr opened the pull request:
Please add "Closes #<issue number>" to the title of the pull request. Then the
issue is closed automatically once it is merged to `main`.

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/main/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/main/articles/unit_test_guidance.html#writing-unit-tests-in-admiral)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/main/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md`
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiralonco/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue so that it closes after successful merging. 
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done!  Much love to your accomplishment!

The following line requests to update the man pages by the "Man Pages" workflow.

/roxygenize

bundfussr commented on vignettes/adtr.Rmd:
````diff
@@ -0,0 +1,599 @@
+)
+```  
+
+## Derive Analysis Flags `ANL02FL`, `ANL03FL`, `ANL04FL` {#analflag}
````
Should we replace "`ANL02FL`, `ANL03FL`, `ANL04FL`" with "(`ANLzzFL`)"?

sgorm123 replied:
Also can we structure this section a little, maybe with sub-sections, just for readability purposes.

HinalPat replied:
done

bundfussr commented on vignettes/adtr.Rmd:
````diff
@@ -0,0 +1,599 @@
+```  
+
+Analysis 04 flag includes all sum of diameters even where lesion identifiers
+match and unmatch if the patient has had PD. Use `CRFL`, `CRNFL`, `PDFL` variables.
````
I think we do not need `CRFL` and `CRNFL` here. (They were already considered for `PDFL`.)

amitjaingsk replied:
we don't use them in our reporting , so its fine to remove CRFL,CRNFL

amitjaingsk replied:
we are keeping them finally. @bundfussr I okay to close this comment, if you are okay , can you close it

bundfussr commented on vignettes/adtr.Rmd:
````diff
@@ -0,0 +1,599 @@
+)
+```  
+
+Analysis 04 flag includes all sum of diameters even where lesion identifiers
+match and unmatch if the patient has had PD. Use `CRFL`, `CRNFL`, `PDFL` variables.
````
Should we replace
>Analysis 04 flag includes all sum of diameters even where lesion identifiers
match and unmatch if the patient has had PD.

with
>Analysis 04 flag includes all sum of diameters where all lesions were assessed and additionally those indicating PD.

?

HinalPat replied:
done added in variables names in ()

bundfussr commented on vignettes/adtr.Rmd:
```diff
@@ -0,0 +1,599 @@
+3. Calculate from source data, the derivation for PD for target lesions is based
+on the calculated sum of diameters of the assessed lesions results in a
+calculated Percent Change from NADIR >=20% and a calculated Change from NADIR
+>=5mm depending on your company specifications.
```
Should we add the complete response criterion (no CR at the current observation and CR at the nadir observation indicates PD)?

amitjaingsk replied:
I think this for data checking purposes for PD to cehck if nadir>=20% etc. do we really need to do here? I would suggest somewhat unconventional to add investigator( or independent reviewer) responses from RS in ADTR for each visit.

HinalPat replied:
we have scenarios where we need to program from scratch.
ours specs have optional RS or ADRS

amitjaingsk replied:
@bundfussr Stefan ,  I am okay to resolve this comment, if its okay for you can resolve

bundfussr commented:
Should we update the `TR` and `TU` test data in admiral.test?

I noticed the following issues:
- Only test code `"DIAMETER"` is available in `TR` but not `"LPERP"` and `"LDIAM"`.
- The location `"LYMHP NODE"` does not occur in `TU`, i.e., all lesions are non-nodal.
- The data is very clean. There are no unscheduled visits or repeated measurements. No assessments which are incomplete but indicate PD.

sgorm123 commented on vignettes/adtr.Rmd:
```diff
@@ -0,0 +1,599 @@
+At this step, it may be useful to join `ADSL` to your `TR` domain.  Only the
+`ADSL` variables used for derivations are selected at this step.  The rest of
+the relevant `ADSL` variables would be added later.
+
```
**_Only the`ADSL` variables used for derivations are selected at this step._**

Can we be a bit more explicit? Only RANDDT is being selected......

sgorm123 commented on vignettes/adtr.Rmd:
```diff
@@ -0,0 +1,599 @@
+should be read into the environment. This will be a company specific process.
+Some of the data frames needed may be `ADSL`, `ADRS` ,`RS`,  `TU` and `TR`.
+
+For example purpose, the SDTM and ADaM datasets (based on CDISC Pilot test
```
Is it _"For example purpose**s**"_?

HinalPat replied:
done

sgorm123 commented on vignettes/adtr.Rmd:
````diff
@@ -0,0 +1,599 @@
+tr <- filter(tr, USUBJID %in% c("01-701-1015", "01-701-1023", "01-701-1028"))
+```
+
+At this step, it may be useful to join `ADSL` to your `TR` domain.  Only the
````
Can we state why is it useful?

HinalPat replied:
done

sgorm123 commented on vignettes/adtr.Rmd:
```diff
@@ -0,0 +1,599 @@
+
+## Derive Parameters TLDn and TLPn {#parameters}
+
+As mentioned earlier, depending on your company specifications you may either
```
I would remove "As mentioned earlier,", I always think this means you have not explained properly earlier.
Just a personal choice.

If we are going to keep it, can we link back to the "mentioned earlier" section.

HinalPat replied:
done

sgorm123 commented on vignettes/adtr.Rmd:
````diff
@@ -0,0 +1,599 @@
+```{r}
+adtr <- tr %>%
+  mutate(
+    tmp_lesion_nr = str_sub(TRLNKID, 3),
````
`str_sub(TRLNKID, 3) `is only repeated twice then you have to remove with a `select(-)`, so I see no benefit in defining. I would just have `str_sub(TRLNKID, 3)` twice, but this is personal preference.

I am sure I read Hadley Wickham state if it repeats at least three times then abstract, two is ok (I may be making this up though).

HinalPat replied:
I leave for @bundfussr to comment

bundfussr replied:
I agree that there is no benefit with respect to the length of the code. However, I think for maintenance it is better not repeating definitions like the lesion number. Then it is easier to adjust the code.
Therefore I have updated the code such that `tmp_lesion_nr` is derived only once.

sgorm123 commented on vignettes/adtr.Rmd:
````diff
@@ -0,0 +1,599 @@
+The examples of this vignette require the following packages. (This is to be
+reviewed/updated once template code is written)
+
+```{r, warning=FALSE, message=FALSE}
````
I notice there are no comments in any of the chunks, it might help with context in places to add some comments.  Like I see in other admiral vignettes, e.g.:

```
# Derive final lab visit date
ds_final_lab_visit <- ds %>%
  filter(DSDECOD == "FINAL LAB VISIT") %>%
  transmute(USUBJID, FINLABDT = convert_dtc_to_dt(DSSTDTC))
# Derive treatment variables
adsl <- dm %>%
  # Merge on final lab visit date
  derive_vars_merged(
    dataset_add = ds_final_lab_visit,
    by_vars = vars(USUBJID)
  )
```

bundfussr replied:
Usually there are no comments in the chunks because the purpose of the chunk is described in the text. I would add comments only if there is something in the code which needs to be explained but is not already explained in the text.

sgorm123 replied:
Ok, then the text above the chunks needs to be updated in several places (particularly one chunk with many dplyr steps).

HinalPat replied:
I have added more comments.

sgorm123 replied:
Great, thank you

HinalPat commented on vignettes/adtr.Rmd:
````diff
@@ -0,0 +1,599 @@
+)
+```
+
+## Derive Parameters TLDn and TLPn {#parameters}
````
Agree!

sgorm123 replied:
Also, TLDn and TLPn, are not created in the chunk, maybe should read `DIAM<n>`

HinalPat replied:
Could we update the template code to include a LDIAMn parameter for lymph nodes diameters and LNDIAMn non-lymph nodes?

HinalPat commented on vignettes/adtr.Rmd:
````diff
@@ -0,0 +1,599 @@
+)
+```
+
+## Derive Parameters (TSD) and Analysis Flag `ANL01FL` {#tsd}
````
Agree!

sgorm123 commented on vignettes/adtr.Rmd:
```diff
@@ -0,0 +1,599 @@
+Depending on your company specifications you may want to merge `TU` with `TR`.
+As an example, here we are keeping the tumor location, and deriving a new
+variable for the tumor group site. You may want to process other variables like
+`ADT`.
```
Can you add the columns being kept and created in this text? i.e.:

Depending on your company specifications you may want to merge `TU` with `TR`.
As an example, here we are keeping the tumor location (**TULOC**), and deriving a new
variable for the tumor group site (**TULOCGR1**). You may want to process other variables like
`ADT`.

HinalPat replied:
done

HinalPat commented on vignettes/adtr.Rmd:
````diff
@@ -0,0 +1,599 @@
+)
+```  
+
+## Derive Baseline (`BASETYPE`, `ABLFL`, `BASE`) {#base}
````
If BASE is calculated, then usually BASETYPE is also populated?

bundfussr replied:
`BASETPYE` is required only if more than one definition of baseline is used within the dataset.

Do we expect this for ADTR?

amitjaingsk replied:
Not sure if basetype is always needed, its needed when we are defining baseline using 2 different derivation.

sgorm123 commented on vignettes/adtr.Rmd:
```diff
@@ -0,0 +1,599 @@
+### Re-labelling unscheduled visits
+
+If you have data collected at unscheduled visits, which are not identified as
+unique in your SDTM or you have company specific rules.  Below is an example of
+how it could be re-labelled. <add more information once I see the template code>
```
My comment is something similar. This chunk is not really handling unscheduled visit slotting.  So maybe just state that it needs to be considered, or add additional information indicating this can be quite complex

HinalPat replied:
I agree Roche derivation is complex.  If we say that grouping lesion diameter by visit is important for the calculations so we only count once, then we should either describe as you say needs to be considered. Maybe with a few details on how this could be approached instead of code?  How do you identify unique visits in the SDTM?

Example from BMS could describe at high level as text ?

Note:  There is a 1-1 relationship between [AVISIT] and [AVISITN]
BMS: Set to -1 if APHASE = "SCREENING".
if APHASE = "POST-BASELINE" then:

Derive temporary variable MAXADY as follows:

(1) For records with EVAL = "INVESTIGATOR": day (ADY) corresponding to the latest date recorded on CRF page ID (the CRF page identifier is recorded as TRSPID/TUSPID variable) in the corresponding period (APHASE).

(2) For records with EVAL = "INDEPENDENT ASSESSOR":

step 1. Find the Maximum date (MAXDATE):

use the latest RS.RSDTC when RSTESTCD = "OVRLRESP" and RSEVAL = "INDEPENDENT ASSESSOR" for corresponding evaluator (EVALID) and visit from RSSPID (If visit is "UNSCHEDULED" merge with " " and 5th substring of RSREFID using "-" as a separator). If BICR response is not available, use the latest ADLE.ADT per ADLE.CRELID for corresponding evaluator (EVALID), where ADLE.EVAL = "INDEPENDENT ASSESSOR".

step 2. Calculate MAXADY as:

MAXADY = MAXDATE - ADCORE.TRTSDT + 1.

If subject was randomized but not treated use ADCORE.RANDDT instead of ADCORE.TRTSDT.

Set to 1 if 2 <= MAXADY <= 11; why not 2 to 8??

set to 2 if 12 <= MAXADY <= 18;

set to 3 if 19 <= MAXADY <= 25;

continue for all other MAXADY values.


Note: AVISITN = x if 7 * x - 2 <= MAXADY <= 7 * x + 4

HinalPat replied:
as per our call. I have left this as it is, explained each company to update based on there requirements

sgorm123 commented on vignettes/adtr.Rmd:
```diff
@@ -0,0 +1,599 @@
+## Derive Parameters (TSD) and Analysis Flag `ANL01FL` {#tsd}
+
+Target Lesions Sum of Diameters is the sum of the diameters for lymph and
+non-lymph nodes (sum of analysis values `AVAL` from source observations TLDn &
```
Is this correct: ```
Target Lesions Sum of Diameters is the sum of the diameters for lymph and
non-lymph nodes```  Just want to double check.

HinalPat replied:
updated sections that were queried here.

sgorm123 commented on vignettes/adtr.Rmd:
```diff
@@ -0,0 +1,599 @@
+TLPn).  The analysis flag `ANL01FL` flags where the number of lesions assessed
+at baseline and at post-baseline match.  To assess whether the number of lesions
+expected and assessed match, you could compare lesion identifiers (`TR.TRLNKID`)
+/use the previous variables set up earlier to compare lesion expected vs lesion
```
what is the `/` ?

HinalPat replied:
removed

sgorm123 commented on vignettes/adtr.Rmd:
````diff
@@ -0,0 +1,599 @@
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
````
Is it me or does this `dataset_vignette`  display slow the RMD down?

bundfussr replied:
That's a known issue. We should restrict the example data to a few patients.

sgorm123 commented on vignettes/adtr.Rmd:
````diff
@@ -0,0 +1,599 @@
+)
+```  
+
+In the below example, the analysis flag, ANL01FL= Y identifies the largest
````
My point regarding the need for comments, would really help in the chunk below.

HinalPat replied:
restructured this bit, some is split now.

sgorm123 commented on vignettes/adtr.Rmd:
```diff
@@ -0,0 +1,599 @@
+
+## Derive Baseline (`BASETYPE`, `ABLFL`, `BASE`) {#base}
+
+These functions are available in `admiral`, below examples show its usage for `ADTR` 
```
Why is this spelt out here? We use lots of functions from admiral.

I was wondering if we need a chunk at the beginning which details exactly the functions we use from admiral, admiral.test, etc.. Given we dont prefix the functions with the package name.

HinalPat replied:
once the template code is final I can add a section at the beginning.

HinalPat replied:
I've mentioned majority are from admiral but kept it at that - consistent with how other vignettes

sgorm123 commented on vignettes/adtr.Rmd:
```diff
@@ -0,0 +1,599 @@
+      new_var = ABLFL,
+      mode = "last"
+    ),
+    filter = ADY <= 1
```
why is this here?

HinalPat replied:
done.

sgorm123 commented on vignettes/adtr.Rmd:
````diff
@@ -0,0 +1,599 @@
+These functions are available in `admiral`, below examples show its usage for `ADTR` 
+
+```{r}
+adtr_sum <- adtr_sum %>%
````
Can we not overwrite each object with the same name (e..g `adtr_sum <- adtr_sum`)? or is this a convention?

I find it hard to track and debug with this,  I guess more for code maintenance than usability.

HinalPat replied:
@bundfussr

bundfussr replied:
For the admiral vignettes we have the convention of using the same dataset name in each step. Thus I would not change it. The convention has the advantage that you can remove or add steps without the need to update the previous and following steps. The downside it that debugging is a bit harder.

sgorm123 commented on vignettes/adtr.Rmd:
````diff
@@ -0,0 +1,599 @@
+```{r}
+adtr_sum <- adtr_sum %>%
+  derive_var_chg() %>%
+  derive_var_pchg() %>%
````
Id a little more information about these functions, i.e. they require AVAL and BASE

HinalPat replied:
done

sgorm123 commented on vignettes/adtr.Rmd:
```diff
@@ -0,0 +1,599 @@
+  derive_var_pchg() %>%
+  mutate(
+    CHGNAD = AVAL - NADIR,
+    PCHGNAD = 100 * CHGNAD / NADIR
```
Should this have an if_else around if NADIR is 0?

bundfussr replied:
I am not sure. `NADIR = 0` results in `PCHGNAD = Inf` for `CHGNAD != 0` and in `PCHGNAD = NaN` for `CHGNAD = 0`.
Alternatively you could set `PCHGNAD = NA` for `NADIR = 0`. I think this is done in most of the admiral functions.

I guess in the SAS transport file all these values are mapped to `.`. Thus it does not make a difference.

sgorm123 commented on vignettes/adtr.Rmd:
````diff
@@ -0,0 +1,599 @@
+`CRNFL`, flags where Complete response contributed to the NADIR calculation.
+
+```{r}
+adtr_sum <- adtr_sum %>% mutate(
````
```
adtr_sum <- adtr_sum %>%  
     mutate(.....
```

sgorm123 commented on vignettes/adtr.Rmd:
```diff
@@ -0,0 +1,599 @@
+
+## Derive Analysis Sequence Number (`ASEQ`) {#aseq}
+
+The `{admiral}` function `admiral::derive_var_obs_number()` can be used to
```
If we are explicit here about admiral functon calls, we should be everywhere, which I think is what we should do. Potentially with a seperate section detaililng all of the external (to admiralonco) function calls.

HinalPat replied:
I copied this section from other vignettes too..seems like in places we are explicit..

sgorm123 commented on vignettes/adtr.Rmd:
```diff
@@ -0,0 +1,599 @@
+tr <- admiral_tr
+rs <- admiral_rs
+
+tu <- convert_blanks_to_na(tu) %>%
+  filter(TUEVAL == "INVESTIGATOR")
+tr <- convert_blanks_to_na(tr) %>%
+  filter(TREVAL == "INVESTIGATOR" & TRGRPID == "TARGET" & TRTESTCD == "DIAMETER")
```
I agree, and also, are we presuming users are aware of each column and xxEVAL, xxGRPID and xxTESTCDs mean. Or should we add a brief description?

HinalPat replied:
Agree, restrict as 'read in' step.  
On top part of the vignette, I mention independent review data - but I can add more details about setting up code twice, unless there is another suggestion?
Also had more comments to the template code.

HinalPat replied:
@amitjaingsk for merging SUPPTU /SUPPTR was there any specific example we could mention?

amitjaingsk replied:
@HinalPat I missed this comment about SUPPxx before, the example you can mention like Tumor laterality info like 'right", "left" info sometime is mapped in SUPPTU. so that can be mentioned as in the text. As its very specific to the study/company, we can mention as a text , otherwise its fine.

HinalPat replied:
@amitjaingsk I have now left it out without the example

sgorm123 commented on vignettes/adtr.Rmd:
```diff
@@ -0,0 +1,599 @@
+* [Merge TR with TU and derive new variables](#merge)
+* [Pre-processing of input records](#input)
+* [Derive Parameters (`TLDn`, `TLPn`)](#parameters)
+* [Derive Parameters (TSD) and ANL01FL](#tsd)
```
**[From Catherine Jin via Stephen Gormley]**

`[Derive Parameters (TSD) and ANL01FL]` is for the general data structure:

1.	 CDISC has instruction to have PARAMCD and PARAM in pair with 1-1 mapping, current PARAMCD=’TSD’ is general on tumor evaluator while PARAM ‘Target Lesions Sum of Diameters by Investigator’ indicates investigator, which could cause issue when company needs both INV and ICR in ADTR.

2.	 PARAM and PARCATx is many-one relation, and only one PARCATx should associate with a given PARAMCD.
So I suggest introducing PARQUAL (Breast Cancer TAUG and Prostate Cancer TAUG both used PARQUAL for evaluator) instead of PARCAT2 and remove “by Investigator” in PARAM.

HinalPat replied:
I noticed that ADRS also has same conventions. How about keeping TSD for investigator, then for IRF, a new parameter name e.g. TSDIRF or whatever company convention is?

We have the same scenario is ADRS, was this discussed before? Anyone recall?

sgorm123 commented on vignettes/adtr.Rmd:
```diff
@@ -0,0 +1,599 @@
+    PARCAT2 = "Investigator",
+    PARCAT3 = "Recist 1.1",
+    AVAL = TRSTRESN,
+    ANL01FL = if_else(!is.na(AVAL), "Y", NA_character_)
```
**[From Catherine Jin via Stephen Gormley]**

This ANL01FL derivation does not consider PD or any new anti-cancer therapy, also does not exclude any pre-baseline records.  Does this require some notification that companies should update the code for this specific ANL01FL derivation (given this flag is the base for several other parameters). Perhaps add something in the header before the chunk, e.g. _depending on your company specifications......_

HinalPat replied:
Hi Catherine, could you explain a bit more about excluding pre-baseline records? 
We discussed to keep ANL01FL as basic then building up on the derivation, ANL03/4FL considers PD.  We are talking about new -anti cancer therapy so perhaps we could tie this in as another example later once we have discussed issue #17 ?  I can explain a bit more in the text about how we approached it, but I saw these as more examples and for each company to update the code based on specific needs.

sgorm123 commented on vignettes/adtr.Rmd:
```diff
@@ -0,0 +1,599 @@
+In the below example, the analysis flag, ANL01FL= Y identifies the largest
+target lesion sum of diameter on a particular date for reporting if there were
+multiple observations on the same date and where lesion assessed at
+post-baseline match baseline.
```
**[From Catherine Jin via Stephen Gormley]**

For cases like PD due to two lesion merges, post-baseline total number of lesions is fewer than baseline. Should we consider more complex algorithms for ANL01FL to cover such situations (and others)?  For example, looking into timepoint overall response at that visit and comparing SUMDIAM with previous visit.

amitjaingsk replied:
As such split and merge of target lesions are not very common, but we should provide some solution. It all depends how the SDTM mapping is done around split/merged lesions. For merging I am seen, T0X/T0Y in TULNKID/TRLNKID and TUTESTCD=MERGED. But this will reduce 1 lesions number as compared to baseline. and for split it TUGRPID will have original lesion number but extra record ( lesion), so again baseline will not match.
So, only counting number of lesions where we have split/merge if lesions will not give us correct results alone. In that case RS target lesion approach is really works well.
The other point of few we already discussed was we are doing SUM irrespective of lesions matching or not. we are creating the SUM record. To consider that SUM for any analysis will be handled using flags later on and each company can define with there approaches and data collection/mapping.

HinalPat replied:
Would be useful to understand a bit more how this alters current way

sgorm123 commented on vignettes/adtr.Rmd:
````diff
@@ -0,0 +1,599 @@
+```{r, echo=FALSE}
+dataset_vignette(
+  arrange(adtr_sum, USUBJID, AVISITN),
+  display_vars = vars(USUBJID, AVISIT, AVAL, NADIR, CHG, CHGNAD, PCHGNAD)
````
**[From Catherine Jin via Stephen Gormley]**

Can we add BASE and PCHG to the display?

HinalPat replied:
done

sgorm123 commented on vignettes/adtr.Rmd:
```diff
@@ -0,0 +1,599 @@
+Below are some examples
+
+Analysis 02 Flag only includes the visit with the lowest NADIR (lowest sum of
+diameter) compared across all visits.  Use PCHGNNAD or PCHG?
```
**[From Catherine Jin via Stephen Gormley]**

Lowest sum of diameter and lowest NADIR both can locate minimum increase and maximum decrease, per my experience lowest sum of diameter PCHG is more commonly used to derive this flag that is used for best tumor shrinkage.

amitjaingsk replied:
agree to use pchg

HinalPat replied:
Lets stick with pchg

sgorm123 commented on vignettes/adtr.Rmd:
````diff
@@ -0,0 +1,599 @@
+```{r eval=TRUE}
+adtr <- adtr %>%
+  derive_vars_merged(
+    dataset_add = select(adsl, !!!negate_vars(adsl_vars)),
````
The adsl_vars object still only contains RANDDT, as in line 149:
`adsl_vars <- vars(RANDDT)`

Should we update this to include other columns from adsl, e.g. SEX, DOB, TRTSTDT

bundfussr replied:
The call adds all ADSL variables to `adtr`. `RANDDT` is excluded because it is already in `adtr`.

amitjaingsk commented on vignettes/adtr.Rmd:
```diff
@@ -0,0 +1,599 @@
+Based on your analysis, additional analysis flags could be set up in `ADTR`.
+Below are some examples
+
+Analysis 02 Flag only includes the visit with the lowest NADIR (lowest sum of
```
we use lowest pchg post baseline ( for max. reduction). sometimes we also create a new record. I think this can be suggested as a text.

amitjaingsk reviewed (commented):
few suggestions/comments

HinalPat commented on vignettes/adtr.Rmd:
```diff
@@ -0,0 +1,599 @@
+
+The main part in programming a tumor results dataset is the calculation of the
+sum of diameters of all target lesions (lymph nodes & non-lymph nodes), the
+calculation of nadir, change in & percentage change from baseline, and the
```
Will update!

HinalPat replied:
done

HinalPat commented on vignettes/adtr.Rmd:
````diff
@@ -0,0 +1,599 @@
+library(lubridate)
+library(stringr)
+library(admiralonco)
+```
````
@bundfussr has this been considered before elsewhere?

bundfussr replied:
The admiral convention is not prefixing the package names.

In the rendered version of the vignette for each function call a link to its documentation is provided. So it is quite easy to see to which package the function is belonging to.

HinalPat replied:
For common functions I have copied from other vignette there is a bit of a mix, inconsistent but consistent across!

sgorm123 replied:
Ok, no worries

HinalPat commented on vignettes/adtr.Rmd:
```diff
@@ -0,0 +1,599 @@
+Depending on your company specifications you may want to merge `TU` with `TR`.
+As an example, here we are keeping the tumor location, and deriving a new
+variable for the tumor group site. You may want to process other variables like
+`ADT`.
```
One example from BMS, they use trdtc, if its missing they use tudtc.  I can expand the text? or remove it all together?

HinalPat replied:
@irgsandy could you add more details about how your approach processing of ADT?

irgsandy replied:
BMS uses trdtc; if it is missing then use tudtc.

For merging TR and TU:
For investigators: 
a.usubjid=b.usubjid_tu and a.TRREFID=b.TUREFID and a.TRLNKID=b.TULNKID;
                  
For BICR:
a.usubjid=b.usubjid_tu and a.trevalid=b.tuevalid and a.TRLNKID=b.TULNKID;

HinalPat replied:
as agreed. I've just added text to explain the date can be processed further if its a consideration in your specs.

HinalPat commented on vignettes/adtr.Rmd:
```diff
@@ -0,0 +1,599 @@
+In this example, we want to flag when a patient has had PD.  This could be done
+in a number of ways:
+
+1. take the date of PD from `ADRS`
```
I think it would be useful, from what I hear each company uses one approach or the other

HinalPat commented on vignettes/adtr.Rmd:
```diff
@@ -0,0 +1,599 @@
+tr <- admiral_tr
+rs <- admiral_rs
+
+tu <- convert_blanks_to_na(tu) %>%
```
I was thinking it would be good to add in - but I notice the other vignettes do not, seems to be only on the get started page on admiral, is there a reason not to repeat it?

bundfussr replied:
We are calling `convert_blanks_to_na()` in the templates but in the vignettes it is not used consistently. I have set it on the agenda of the admiral devs meeting.

HinalPat replied:
added a link to main admiral page

sgorm123 replied:
sounds good

HinalPat commented:
@bundfussr I've made updates based on comments received, and built the vignette and checked how it looks. Please see additional test data, and potential template code updates below:

**Updates to test data:**

- Include some missing lesion measurements and where lesion size is missing when we display the various new variables, and and some PD scenarios so that the analysis flags later user can see how this impacts the ANL01FL vs ANL02/03/04FL
- Include Missing day in TRDTC so we can see ADTF populated
- name parameters as per vignette

**template code:**
- Create template code for both LDIAM and NLDIAM
- SDIAM is then a sum of both LDIAM and NLDIAM
- when we call derive_var_merged_summary for ADY and ADT, I'm not sure if its doing as per vignette text as want the maximum lesion size not min date, and not sure if we need function for ADY and ADT?
- the orderings of the variables on the data_vignette when I display the vignette doesn't match up nicely- can this be fixed? when we display % change in NADIR its doesn't fit - possible to fix without dropping anything?
- these some code adjustments mentioned earlier in the pull request I won't repeat these, please take a look and feedback.
- ANL02FL flag should look at records where analysis date after randomization date
- ANL03FL PDFL=Y and so does ANL03FL=Y but should be missing?
- ANL04FL include a scenario where ANL01FL=is missing but ANL04FL=Y

**Vignette:**
- Some links don't work could you help fix them - then I can see how it needs to be done!

bundfussr commented:
> * Some links don't work could you help fix them - then I can see how it needs to be done!

@HinalPat , I found one link ("Derive Parameter for Sum of Diameter and Analysis Flag `ANL01FL`") which did not work and fixed it (the anchor was uppercase but the link used lower case). Are there any other links not working?

bundfussr commented:
> * when we call derive_var_merged_summary for ADY and ADT, I'm not sure if its doing as per vignette text as want the maximum lesion size not min date, and not sure if we need function for ADY and ADT?

@HinalPat , we derive `ADT` and `ADY` because the lesions could have been assessed at different days within a visit. We need to select one of them as date/day of the sum. In the example we use the first one. Maybe we should display it in a separate code chunk and add some text. What do you think?

bundfussr commented:
> * the orderings of the variables on the data_vignette when I display the vignette doesn't match up nicely- can this be fixed?

@HinalPat , I am not sure regarding the ordering of the variables. For which chunks it is not as expected?

> when we display % change in NADIR its doesn't fit - possible to fix without dropping anything?

It is not possible to adjust the columns widths. Thus you need to scroll horizontally to see all variables for % change in nadir. I think it's better than dropping any of the variables.

HinalPat commented:
> > * Some links don't work could you help fix them - then I can see how it needs to be done!
> 
> @HinalPat , I found one link ("Derive Parameter for Sum of Diameter and Analysis Flag `ANL01FL`") which did not work and fixed it (the anchor was uppercase but the link used lower case). Are there any other links not working?

@bundfussr theres a couple under 'Read data', and line 273

HinalPat commented:
> > * when we call derive_var_merged_summary for ADY and ADT, I'm not sure if its doing as per vignette text as want the maximum lesion size not min date, and not sure if we need function for ADY and ADT?
> 
> @HinalPat , we derive `ADT` and `ADY` because the lesions could have been assessed at different days within a visit. We need to select one of them as date/day of the sum. In the example we use the first one. Maybe we should display it in a separate code chunk and add some text. What do you think?

@bundfussr In the text before, we say for duplicate dates take the largest lesion sum. I setup a call to discuss this one.

HinalPat commented:
> @HinalPat , I think I have addressed all of your comment. I hope I did not miss anything. Otherwise, please let me know.
> 
> I have also modified the test data to add some incomplete dates. I.e., you need to call
> 
> ```
> remotes::install_github("pharmaverse/admiral.test", ref = "103_update_tr@devel")
> ```
> 
> and restart before building the vignette.

@bundfussr I am unable to rerun vignette build, I get the error message:
Error in exprs(RANDDT) : could not find function "exprs"

Once I can rerun, will check for ordering of data_vignette variables and check your updates.

bundfussr commented:
> > > * Some links don't work could you help fix them - then I can see how it needs to be done!
> > 
> > 
> > @HinalPat , I found one link ("Derive Parameter for Sum of Diameter and Analysis Flag `ANL01FL`") which did not work and fixed it (the anchor was uppercase but the link used lower case). Are there any other links not working?
> 
> @bundfussr theres a couple under 'Read data', and line 273

I have fixed the links.

HinalPat commented:
@bundfussr Hi Stefan please find my comments:

- line 307 - AVAL on the dataset vignette display is presented at 3rd column - when we want it to show up at the end
- line 345-365 could you split up and then move comment on line 321 above the ADT/ADY bit.
- line 421 - AVAL should come up before nadir on the dataset vignette but is 3rd column
- Nadir calculation, pt 1015 wk6 be 77.1?
- PDFL flags, I think in the example its abit confusing when PDFL=Y at week 6 and week 24 generally once patient has PD response shouldn't change
- Line 595, ANL01FL not sitting with ANL02FL in the vignette build, same for ANL01FL/ANL03FL and ANL01FL/ANL04FL
thank you!

bundfussr commented:
> @bundfussr Hi Stefan please find my comments:
> 
> * line 307 - AVAL on the dataset vignette display is presented at 3rd column - when we want it to show up at the end
> * line 345-365 could you split up and then move comment on line 321 above the ADT/ADY bit.
> * line 421 - AVAL should come up before nadir on the dataset vignette but is 3rd column
> * Nadir calculation, pt 1015 wk6 be 77.1?
> * PDFL flags, I think in the example its abit confusing when PDFL=Y at week 6 and week 24 generally once patient has PD response shouldn't change
> * Line 595, ANL01FL not sitting with ANL02FL in the vignette build, same for ANL01FL/ANL03FL and ANL01FL/ANL04FL
>   thank you!

@HinalPat , everything should be fixed. `PDFL` differs for the different options because in ADRS and RS non-target lesions are taken into account for PD.

HinalPat commented:
>everything should be fixed. PDFL differs for the different options because in ADRS and RS non-target lesions are taken into account for PD.

@bundfussr for PD we should only consider the first date of PD, then PDFL=Y where first date is identified, so we only see one Y?

bundfussr commented:
@HinalPat , I have removed the assessments after PD from the vignette.

amitjaingsk commented on vignettes/adtr.Rmd:
```diff
@@ -0,0 +1,725 @@
+flags therefore should be setup according to your company specifications.
+
+
+In the example below, the lesion sizes are counted and grouped according to the by variables specified, in this case we group by the visit (`AVISIT` and `AVISIT`) to ensure that the correct lesion measurements are summed together. 
```
AVISIT is coming 2 times, I think you mean to say AVISITN?

bundfussr replied:
Fixed

amitjaingsk commented on vignettes/adtr.Rmd:
```diff
@@ -0,0 +1,725 @@
+times in order to process additional variables for sum of diameter parameter,
+and calculate the analysis flag (`ANL01FL`).
+
+In the first step we concatenate lesions IDS measured `LSEXP` at baseline, and merge in with the second step where we concatenate the lesions IDS assessed `LSASS` at that visit record.  You can then compare `LSEXP` vs `LSASS` at each visit.
```
Minor change IDS -> IDs

bundfussr replied:
Fixed

amitjaingsk commented on vignettes/adtr.Rmd:
```diff
@@ -0,0 +1,725 @@
+
+## Derive Baseline (`ABLFL`, `BASE`) {#base}
+
+These functions are available in `admiral`, below examples show its usage for `ADTR`. Here we create baseline record flag `ABLFL` by selecting all records prior to analysis day 1, and then derive the baseline value `BASE`.
```
As for baseline we are assuming all lesions at ady<=1 are assessed are at baseline visit. But I have seen some time lesions are also recorded at screening. So, I suggest to add a statement in the end " if in a particular study/or subject, lesions are also collected at multiple visits before day 1, so make sure to update appropriate filter condition."

bundfussr replied:
The text was not 100% precise. The function call does not select _all_ records but the _last_ record prior to analysis day 1. I have updated the paragraph.

Please have a look.

amitjaingsk commented on vignettes/adtr.Rmd:
```diff
@@ -0,0 +1,725 @@
+adtr_sum <- adtr_sum %>% mutate(
+  CRFL = if_else(AVAL == 0 & ANL01FL == "Y", "Y", NA_character_),
+  CRNFL = if_else(NADIR == 0, "Y", NA_character_),
+  PDFL = if_else(is.na(CRFL) & CRNFL == "Y", "Y", NA_character_)
```
why we are again creating PDFL here, when we already provided example derviation for PDFL above

bundfussr replied:
For `PDFL` we show four options how it could be derived. The users would need to select one of them.

amitjaingsk reviewed (commented):
I have some minor comment/suggestions..

bundfussr commented on vignettes/adtr.Rmd:
```diff
@@ -0,0 +1,725 @@
+explains why we need to use the `convert_blanks_to_na()` function.
+
+[^1]: In this vignette we use only data for subjects "01-701-1015",
+"01-701-1023", "01-701-1440".
```
We use this restriction only to avoid that the page needs a long time to load. You wouldn't use it in a real study.

sgorm123 commented on vignettes/adtr.Rmd:
```diff
@@ -0,0 +1,734 @@
+The main part in programming a tumor results dataset is the calculation of the
+sum of diameters of all target lesions (lymph nodes & non-lymph nodes), the
+calculation of nadir, change & percentage change from baseline, and the
+analysis flags that could be required for reporting.  The tumor results data could be set up for investigator and/or Independent review facility (IRF)/Blinded Independent Central Review (BICR) data.  The below sample code would need to be updated (for example, update the Evaluator `TR.TREVAL`, `TU.TUEVAL`, and the applicable parameter details `PARAM`, `PARAMCD`, `PARCATx`) in order to create the  ...
```
appear to be a few erroneous line breaks here (line 52, 53 and 54)

bundfussr replied:
It doesn't matter if you have line breaks within a paragraph or if the whole paragraph is written in one line. On the website it is displayed correctly.

sgorm123 commented on vignettes/adtr.Rmd:
````diff
@@ -0,0 +1,599 @@
+
+```{r echo=FALSE}
+tr <- filter(tr, USUBJID %in% c("01-701-1015", "01-701-1023", "01-701-1028"))
+```
````
So this is ok as is, no problem

irgsandy commented on vignettes/adtr.Rmd:
```diff
@@ -0,0 +1,599 @@
+As mentioned earlier, depending on your company specifications you may either
+create parameters for TLDn and TLPn in ADTR or similar dataset.
+
+Set the `AVALx/ANL01FL` variables.
```
Same question here. What's AVALx? If there are different values, maybe we can use paramcd to differentiate.

HinalPat replied:
x is removed

HinalPat commented on vignettes/adtr.Rmd:
````diff
@@ -0,0 +1,599 @@
+```{r echo=FALSE}
+# work-around for https://github.com/pharmaverse/admiral.test/issues/98
+# should be removed once the bug is fixed
+tr <- ungroup(tr)
````
this is now removed

HinalPat commented on vignettes/adtr.Rmd:
```diff
@@ -0,0 +1,599 @@
+at baseline and at post-baseline match.  To assess whether the number of lesions
+expected and assessed match, you could compare lesion identifiers (`TR.TRLNKID`)
+/use the previous variables set up earlier to compare lesion expected vs lesion
+accessed  or check for target lesion response (from `RS`) =NE.
```
done

HinalPat commented on vignettes/adtr.Rmd:
````diff
@@ -0,0 +1,599 @@
+```{r}
+adtr_sum <- get_summary_records(
+  adtr,
+  by_vars = vars(STUDYID, USUBJID, !!!adsl_vars, AVISIT, AVISITN),
````
done

HinalPat commented on vignettes/adtr.Rmd:
```diff
@@ -0,0 +1,599 @@
+
+## Pre-processing of input records {#input}
+
+The next step involves company-specific pre-processing of records for the
```
done

HinalPat commented on vignettes/adtr.Rmd:
```diff
@@ -0,0 +1,599 @@
+## Derive `NADIR` {#nadir}
+
+Once we have a TSD parameter, the NADIR is set to the lowest sum of diameters in
+Analysis Value before the current observation, where Analysis Flag 01 =Y.  The
```
done

HinalPat commented on vignettes/adtr.Rmd:
```diff
@@ -0,0 +1,599 @@
+Once we have a TSD parameter, the NADIR is set to the lowest sum of diameters in
+Analysis Value before the current observation, where Analysis Flag 01 =Y.  The
+first observations after baseline (depending on how you have defined baseline
+earlier) is set to analysis value where baseline record flag = Y
```
_done_

HinalPat commented on vignettes/adtr.Rmd:
```diff
@@ -0,0 +1,599 @@
+
+These functions are available in admiral, below examples show its usage for
+`ADTR`.  We calculate the change from baseline and percentage change from
+baseline for the parameter TSD and the variable `NADIR`.
```
done

HinalPat commented on vignettes/adtr.Rmd:
```diff
@@ -0,0 +1,599 @@
+flag variables that will then aid setting up any additional analysis flag
+variables defined as per your company specifications.
+
+###  Derive `CRFL`, `CRNFL`
```
done

HinalPat commented on vignettes/adtr.Rmd:
```diff
@@ -0,0 +1,599 @@
+
+In addition, you could create any additional new variables that are required
+downstream when deriving the required parameters. For example, here we include
+lesion ID expected and lesion ID assessed. (any other examples?)
```
I leave code comments for @bundfussr to comment

bundfussr replied:
@irgsandy , where would you move the `mutate()` call to?
